cookbooks/bcpc/recipes/keepalived.rb

```diff
@@ -20,7 +20,23 @@
 include_recipe "bcpc::default"
 
 make_config('keepalived-router-id', "#{(rand * 1000).to_i%254/2*2+1}")
-make_config('keepalived-password', secure_password)
+
+keepalived_password = get_config('keepalived-password')
+if keepalived_password.nil?
+  keepalived_password = secure_password
+end
+
+bootstrap = get_all_nodes.select{|s| s.hostname.include? 'bootstrap'}[0].fqdn
+
+nodes = get_nodes_for("keepalived").map!{ |x| x['fqdn'] }.join(",")
```

Comment:
get_node_attributes returns a array of hash with the requested attributes. It needs more logic to extract what I need from the hash array and reassemble them into a string. Extra-complexity will be added. So simple map() seems not a bad idea.